Add tests to Subscriptions inclusion

octaedro opened the pull request:
This is a follow-up to [PR 7777](https://github.com/woocommerce/woocommerce-admin/pull/7777) to add tests for the Subscriptions inclusion.

### Detailed test instructions:

Run unit and e2e tests and verify that everything is working well.

<!--
Please add your test instructions to `/TESTING-INSTRUCTIONS.md`.
-->

<!--- Please add a Changelog note

Enter a changelog note using the following CLI command `npm run changelogger -- add` and commit changes. --->

octaedro committed:
Add unit tests to `OnboardingProductTypes` REST API

octaedro committed:
Add e2e tests

octaedro committed:
Add changelog

octaedro committed:
Fix e2e tests

octaedro committed:
Fix test description

octaedro committed:
Fix to clean payment methods

louwie17 commented on packages/admin-e2e-tests/src/pages/ProductsSetup.ts:
```diff
@@ -0,0 +1,30 @@
+			'span',
+			'Start with a template'
+		);
+		await startWithATemplate?.click();
```
You could make use of the `this.clickElementWithText` that you added to the `BasePage` right?
Also we might want to use a wildcard `*` instead of `span`

octaedro replied:
Nice catch! I added those changes in commit 9ca1bdd

louwie17 commented on packages/admin-e2e-tests/src/specs/activate-and-setup/complete-onboarding-wizard.ts:
```diff
@@ -349,8 +351,175 @@ const testDifferentStoreCurrenciesWCPay = () => {
+			await profileWizard.continue();
+		} );
+
+		it( 'cannot see the WooCommerce Payments extension after it has been installed', async () => {
```
This expect here is expecting `WooCommerce Payments` to show up, but the test description say's otherwise?
I assume it should show `WooCommerce Payments` here for a non-us store.

octaedro replied:
You're right, I fixed the description in commit 982450c

louwie17 commented on packages/admin-e2e-tests/src/utils/actions.ts:
```diff
@@ -167,6 +168,23 @@ export const waitForElementByTextWithoutThrow = async (
 	return Boolean( selected );
 };
 
+const deactivateAndDeleteExtension = async ( extension: string ) => {
```
Nice!

louwie17 reviewed (commented):
@octaedro thanks so much for writing some tests for this, especially E2E tests 👍 👍 

I left a couple minor comments, and we probably also want to add a changelog to the `admin-e2e-tests` library. Maybe two separate items, one referring to the extension deactivation util function addition and the other for adding additional subscription tests.

octaedro committed:
Fixed `clickStartWithTemplate`

octaedro committed:
Fix test description

octaedro committed:
Add package changelog

octaedro commented:
Thank you @louwie17 for your review.
I added the suggested changes. I also added the changelog entries in commit 7b4dcab. 
Could you take another look at this PR?

louwie17 reviewed (approved):
Thanks for the updates @octaedro, everything looks good!
There was one E2E test failure, but not related to your changes, that one seems to fail off and on (I need to address that).
